src/main/java/teammates/logic/api/Logic.java

```diff
@@ -2622,4 +2622,14 @@ public MimeMessage emailErrorReport(String path, String params, Throwable error)
     @SuppressWarnings("unused")
     private void ____helper_methods________________________________________() {
     }
+    
+    public void extractActiveAndArchivedCourses(List<CourseDetailsBundle> courseBundles, 
```

Comment:
When writing this function, I was thinking of saving the for loop, which retrieve information from the Db so that it can reduce the number of reading operations. Is using output parameters worth for that? If not then I think separating them into 2 functions would be better.